src/AutoUI/Filters/PersistentFilters.tsx

```diff
@@ -210,22 +218,39 @@ export const PersistentFilters = ({
 				strictNullHandling: true,
 			});
 
-			history?.replace?.({
-				pathname,
-				search: filterQuery,
-			});
+			if (
+				history != null &&
+				typeof history === 'object' &&
+				'replace' in history &&
+				typeof history.replace === 'function'
+			) {
+				// react-router-dom v5 history object
+				history.replace({
+					pathname,
+					search: filterQuery,
+				});
+			} else if (typeof history === 'function') {
+				// react-router-dom v6 navigate function
+				history(
+					{
+						pathname,
+						search: filterQuery,
+					},
+					{ replace: true },
+				);
+			}
 
 			onFiltersChange?.(updatedFilters);
 
-			if (filterQuery !== locationSearch.substring(1)) {
+			if (filterQuery !== search.substring(1)) {
 				analytics.webTracker?.track('Update table filters', {
 					current_url: location.origin + location.pathname,
 					// Need to reduce to a nested object instead of nested array for Amplitude to pick up on the property
 					filters: Object.assign({}, parsedFilters),
 				});
 			}
 		},
-		[onFiltersChange, analytics.webTracker, history, locationSearch, pathname],
+		[onFiltersChange, analytics.webTracker, history, search, pathname],
```

Comment:
as before, `search` and `pathname` can be removed since `document.location` is not a reactive source